docs/src/main/asciidoc/security-openid-connect-web-authentication.adoc

```diff
@@ -648,9 +664,87 @@ public class TokenResource {
         return userInfo.getUserInfoString();
     }
 }
-```
+----
+
+If you support more than one social provider with the help of link:security-openid-connect-multitenancy[OpenId Connect Multi-Tenancy], for example, `Google` which is an OpenId Connect Provider returning `IdToken` and `GitHub` which is an `OAuth2` provider returning no `IdToken` and only allowing to access `UserInfo` then you can have your endpoint working with only the injected `SecurityIdentity` for both `Google` and `GitHub` flows. A simple augmentation of `SecurityIdentity` will be required where a principal created with the internally generated `IdToken` will be replaced with the `UserInfo` based principal when the GiHub flow is active:
+
+[source,java]
+----
+package io.quarkus.it.keycloak;
+
+import java.security.Principal;
+
+import javax.enterprise.context.ApplicationScoped;
+
+import io.quarkus.oidc.UserInfo;
+import io.quarkus.security.identity.AuthenticationRequestContext;
+import io.quarkus.security.identity.SecurityIdentity;
+import io.quarkus.security.identity.SecurityIdentityAugmentor;
+import io.quarkus.security.runtime.QuarkusSecurityIdentity;
+import io.smallrye.mutiny.Uni;
+import io.vertx.ext.web.RoutingContext;
+
+@ApplicationScoped
+public class CustomSecurityIdentityAugmentor implements SecurityIdentityAugmentor {
+
+    @Override
+    public Uni<SecurityIdentity> augment(SecurityIdentity identity, AuthenticationRequestContext context) {
+        RoutingContext routingContext = identity.getAttribute(RoutingContext.class.getName());
+        if (routingContext != null && routingContext.normalizedPath().endsWith("/github")) {
+	        QuarkusSecurityIdentity.Builder builder = QuarkusSecurityIdentity.builder(identity);
+	        UserInfo userInfo = identity.getAttribute("userinfo");
+	        builder.setPrincipal(new Principal() {
+	
+	            @Override
+	            public String getName() {
+	                return userInfo.getString("preferred_username");
+	            }
+	
+	        });
+	        identity = builder.build();
```

Comment:
This looks like boilerplate code. Can we somehow provide this augmentor in the extension (maybe as a config parameter?)

Reply:
Hi @gastaldi Interesting idea. It is unlikely it will be a `preferred_username` for GitHub (it is used by KC but the augmentor is not needed there) and there is also an application specific path check here (assuming it won't be needed for the Google flow in this case). In general I'd like to avoid having some specific code support for non-OIDC compliant providers like GitHub - we already stretch `quarkus-oidc` a bit to support GitHub flows :-). Though in this specific case I've just typed this code to suggest how to deal with both Google and GitHub at the same time, withh GitHib alone accessing the injected `UserInfo` would be simpler as shown in the example code in the same section. So if you don't mind, I'll look at it later, may be, if at some stage, we'll have some provider specific separate extension helpers then we can have a few of such custom augmentors, etc

Though Pedro had an idea to simplify the config for GitHub, #20783, it would definitely like to pursue that one as it can indeed be generally useful (we'd need to have a notion of the configuration block instances which can be referred to from elsewhere)